libflux/src/flux/semantic/nodes.rs

```diff
@@ -415,11 +457,20 @@ pub struct BuiltinStmt {
 }
 
 impl BuiltinStmt {
-    fn infer(&self, _: Environment, _: &mut Fresher) -> Result {
-        unimplemented!();
+    fn infer<I: Importer>(
+        &self,
+        mut env: Environment,
+        importer: &I,
+    ) -> std::result::Result<Environment, Error> {
+        if let Some(ty) = importer.import(&self.id.name) {
+            env.add(self.id.name.clone(), ty.clone());
+            Ok(env)
+        } else {
+            Err(Error::undefined_builtin(&self.id.name))
```

Comment:
Ok, cool, so the build will fail if we forget to add types for builtins.

Reply:
Yes.